src/patterns/gen-ai/aws-rag-appsync-stepfn-kendra/index.ts

```diff
@@ -0,0 +1,526 @@
+  /**
+     * Optional Name for the Lambda function environment variable set to the index id for the Kendra index.
+     *
+     * @default - KENDRA_INDEX_ID
+     */
+  readonly indexIdEnvironmentVariableName?: string;
+  /**
+   * Optional. Allows to provide Generate Pre-signed Url custom lambda code
+   * and settings instead of the existing
+   */
+  readonly generatePresignedUrlLambdaProps?: DockerLambdaCustomProps | undefined;
+  /**
+   * Optional. Allows to provide Kendra Start Data Sync Job custom lambda code
+   * and settings instead of the existing
+   */
+  readonly kendraStartDataSyncLambdaProps?: DockerLambdaCustomProps | undefined;
+  /**
+   * Optional. Allows to provide Check Sync Job Status custom lambda code
+   * and settings instead of the existing
+   */
+  readonly checkJobsStatusLambdaProps?: DockerLambdaCustomProps | undefined;
+  /**
+   * Optional. Allows to provide Update Kndra Sync Job Status custom lambda code
+   * and settings instead of the existing
+   */
+  readonly updateKendraJobStatusLambdaProps?: DockerLambdaCustomProps | undefined;
+}
+
+enum ServiceEndpointTypeEnum {
+  DYNAMODB= 'DDB',
+  ECR_API= 'ECR_API',
+  ECR_DKR= 'ECR_DKR',
+  EVENTS= 'CLOUDWATCH_EVENTS',
+  KENDRA= 'KENDRA',
+  KINESIS_FIREHOSE= 'KINESIS_FIREHOSE',
+  KINESIS_STREAMS= 'KINESIS_STREAMS',
+  S3= 'S3',
+  SAGEMAKER_RUNTIME= 'SAGEMAKER_RUNTIME',
+  SECRETS_MANAGER= 'SECRETS_MANAGER',
+  SNS= 'SNS',
+  SQS= 'SQS',
+  SSM= 'SSM',
+  STEP_FUNCTIONS= 'STEP_FUNCTIONS',
+}
+
+/**
+ * @summary The RagAppsyncStepfnKendra class.
+ */
+export class RagAppsyncStepfnKendra extends Construct {
+  public readonly vpc?: ec2.IVpc;
+  public readonly kendraIndex: KendraConstruct;
+  public readonly kendraInputBucket: Bucket;
+
+  public readonly graphqlApi: appsync.IGraphqlApi;
+  private readonly kendraIndexId: string;
+  private readonly awsRegion: string;
+  private readonly awsAccountId: string;
+  lambdaTracing: lambda.Tracing=lambda.Tracing.ACTIVE;
+  kendraInputBucketArn: string;
+  public readonly securityGroup: ec2.ISecurityGroup;
+
+  /**
+     * Returns an instance of appsync.IGraphqlApi created by the construct
+     */
+  // @ts-ignore
+  docProcessingStateMachine: StateMachine;
+  // @ts-ignore
+  stepFunctionsExecutionHandlerRole: cdk.aws_iam.Role;
+  // @ts-ignore
+  syncRunTable: Table;
+  //   TODO(miketran): Right now, we support S3, then eventually will open it up to other data sources.
+  private kendraDataSourceIndexId: string;
+  private stack: Stack;
+  cognito: any;
+
+
+  /**
+     * @summary Constructs a new instance of the RagAppsyncStepfnKendra class.
+     * @param {cdk.App} scope - represents the scope for all the resources.
+     * @param {string} id - this is a scope-unique id.
+     * @param {RagAppsyncStepfnKendraProps} props - user provided props for the construct.
+     * @since 0.0.0
+     * @access public
+   */
+  constructor(scope: Construct, id: string, props: RagAppsyncStepfnKendraProps) {
```

Comment:
Can we add support for metrics tracking, similar to:
https://github.com/awslabs/generative-ai-cdk-constructs/blob/main/src/patterns/gen-ai/aws-contentgen-appsync-lambda/index.ts#L174C5-L180C7